spec/src/main/asciidoc/images/lra-state-model.txt

```diff
@@ -56,6 +56,8 @@ state LRAContext {
   ActiveLRA --> Closing : close
   Closing --> Closed
   Closing --> FailedToClose
+
+  Closed --> Cancelling : cancel (transition is only allowed if LRA is nested)
```

Comment:
How does a nested participant know that it can release resources? It seems it cannot release resources when it is closed so I presume some other callback is issued but I don't see such a notion mentioned in 3.2.7

Reply:
I agree. We can require that nested LRA participants must declare Forget methods. In that sense:
* if nested LRA closes, its participants must keep resources (same as current behavior)
* if nested LRA cancels, its participant can release resources
* if parent LRA closes, Forget method must be called even on nested LRA participants

Reply:
`@AfterLRA` could be used

Reply:
The participant knows whether or not he is executing in a nested context.

If he is enlisted with a nested LRA then he would hold onto any required state until it got the AfterLRA notification.
If he is enlisted with a top level LRA then he would release any required state when he has processed the compensate or complete notification correctly.

Reply:
Okay let's go with `@Forget` then. If there are no other opinions then I will push a spec update.

Reply:
+1 to `@Forget`

Reply:
@tomjenkinson and @xstefank I have added some clarifying remarks about the use of `@Forget`